# Jupyter Notebook For Unit 6

## Workbook 6

In [69]:
import pandas as pd
import pprint

file_path = 'data/Exe8.2B.xlsx'  # Update this to the path of your Excel file
sheet_name = 'Diets'  # Update this with the name of your sheet if needed
df = pd.read_excel(file_path, sheet_name=sheet_name)
# Specify the columns
diet_cl = 'Diet'
weight_loss_cl = 'Wtloss'
pp = pprint.PrettyPrinter(indent=4)
pp.pprint(df[[diet_cl, weight_loss_cl]])

   Diet  Wtloss
0     A   3.709
1     A   7.087
2     A   6.754
3     A   8.994
4     A   9.077
..  ...     ...
95    B   3.972
96    B   2.409
97    B   0.569
98    B   7.013
99    B   2.594

[100 rows x 2 columns]


# Diet A Calculations

In [70]:
# Filter the DataFrame where Diet has "Value A" and Weight loss columns value
filtered_df_A = df.loc[(df[diet_cl] == 'A'), weight_loss_cl ]

mean_A = filtered_df_A.mean().round(3)
std_A = round(filtered_df_A.std(), 3)
med_A = round(filtered_df_A.median(),3)
pp.pprint(f"Total Number of Diet A: {len(filtered_df_A)},Mean of Diet A: {mean_A}, Standard Deviation of Diet A: {std_A}")

q1_A = round(filtered_df_A.quantile(0.25),3)
q3_A = round(filtered_df_A.quantile(0.75),3)
iqr_A = round(q3_A-q1_A,3)
pp.pprint(f"Median Diet A: {med_A}, Q1:{q1_A}, q3:{q3_A}, IQR:{iqr_A}")

('Total Number of Diet A: 50,Mean of Diet A: 5.341, Standard Deviation of Diet '
 'A: 2.536')
'Median Diet A: 5.642, Q1:3.748, q3:7.033, IQR:3.285'


# Diet B calculations

In [71]:
# Filter the DataFrame where Diet has "Value B" and Weight loss columns value
filtered_df_B = df.loc[(df[diet_cl] == 'B'), weight_loss_cl ]

mean_B = filtered_df_B.mean().round(3)
std_B = round(filtered_df_B.std(), 3)
med_B = round(filtered_df_B.median(),3)
pp.pprint(f"Total Number of Diet B: {len(filtered_df_B)}, Mean of Diet B: {mean_B}, Standard Deviation of Diet B: {std_B}")
q1_B = round(filtered_df_B.quantile(0.25),3)
q3_B = round(filtered_df_B.quantile(0.75),3)
iqr_B = round(q3_B-q1_B,3)
pp.pprint(f"Median Diet B: {med_B}, Q1:{q1_B}, q3:{q3_B}, IQR:{iqr_B}")

('Total Number of Diet B: 50, Mean of Diet B: 3.71, Standard Deviation of Diet '
 'B: 2.769')
'Median Diet B: 3.745, Q1:1.953, q3:5.404, IQR:3.451'


## Results
### Results: Diet A (Sample Size =50)

| Mean  | Median | Standard Deviation | Q1    | Q3    | IQR   |
|-------|--------|--------------------|-------|-------|-------|
| 5.341 | 5.642  | 2.536              | 3.748 | 7.033 | 3.285 |

### Result: Diet B (Sample Size =50)

| Mean | Median | Standard Deviation | Q1    | Q3    | IQR   |
|------|--------|--------------------|-------|-------|-------|
| 3.71 | 3.745  | 2.769              | 1.953 | 5.404 | 3.451 |

### Exercise 6.1

<img src="data/ex6-1.png" width="1200" height="800" />

Looking at the Standard deviations of both Diet A (2.536) and Diet B (2.769) the deviations of the weight loss from are very similar and both data diets exhibit similar variability, However, the mean for Diet A shows a better weight loss of 1.6 kg on average on this sample size of 50, which suggests that Diet A should be preferred. 

### Exercise 6.2

<img src="data/ex6-2.png" width="1200" height="800" />

Looking at the sample interquartile range (IQR) for Diet A(3.285) and Diet B (3.451), it can been observed that IQR for Diet B is a bit higher than of Diet A, suggesting that Diet B has is more inconsistent as it has more varying data compared to Diet A. To conclude, Diet A presents a more consistent and effective weight loss compared to Diet B, as the median is higher for Diet A presenting higher effectiveness and more consistent because of less variability due to narrower IQR.  




In [72]:
import pandas as pd
import pprint

file_path = 'data/Exe8.3D.xlsx'  # Update this to the path of your Excel file
sheet_name = 'Sheet1'  # Update this with the name of your sheet if needed
df = pd.read_excel(file_path, sheet_name=sheet_name)
# Specify the columns
area_cl = 'Area'
brand_cl = 'Brand'
pp = pprint.PrettyPrinter(indent=4)
pp.pprint(df[[area_cl, brand_cl]])

# Filter the DataFrame where Diet has Area 2
filtered_df_area_1 = df.loc[(df[area_cl] == 2), brand_cl ]

     Area  Brand
0       1      B
1       1  Other
2       1      A
3       1      B
4       1  Other
..    ...    ...
155     2      A
156     2      B
157     2      B
158     2  Other
159     2  Other

[160 rows x 2 columns]


## Frequencies Calculation


In [73]:

filtered_brand_A = filtered_df_area_1.value_counts()['A']
filtered_brand_B = filtered_df_area_1.value_counts()['B']
filtered_brand_other = filtered_df_area_1.value_counts()['Other']
total =len(filtered_df_area_1)
pp.pprint(f"A: {filtered_brand_A}, B:{filtered_brand_B}, Other:{filtered_brand_other}, Total:{total}")

'A: 19, B:30, Other:41, Total:90'


## Frequency Percentage

In [74]:
filtered_brand_A_per = (filtered_df_area_1.value_counts()['A']/total)*100
filtered_brand_B_per = (filtered_df_area_1.value_counts()['B']/total)*100
filtered_brand_other_per = (filtered_df_area_1.value_counts()['Other']/total)*100
total_per =filtered_brand_A_per + filtered_brand_B_per + filtered_brand_other_per
pp.pprint(f"A %: {filtered_brand_A_per}, B%:{filtered_brand_B}, Other%:{filtered_brand_other_per}, Total%:{total_per}")

'A %: 21.11111111111111, B%:30, Other%:45.55555555555556, Total%:100.0'


## Results
### Frequencies for Area 2

| A  | B  | Other | Total |
|----|----|-------|-------|
| 19 | 30 | 41    | 90    |

### Frequencies Percentage

| A     | B     | Other | Total |
|-------|-------|-------|-------|
| 21.11 | 33.33 | 45.56 | 100   |


### Exercise 6.3

<img src="data/ex6-3.png" width="1200" height="800" />
<br>
Looking at the frequency for Area 1 and Area 2, both of the areas prefer some Other brand of Cereal with Area 1 having 60% and Area 2 having 45.6%. However, larger percentage of people from Area 1 preferred the other brand in comparison to Area 2. 

